app/routes/surveys/components/SurveyEditor/Question.js

```diff
@@ -0,0 +1,167 @@
+// @flow
+
+import React from 'react';
+import { type FieldArrayProps, Field, FieldArray } from 'redux-form';
+import {
+  TextInput,
+  TextArea,
+  CheckBox,
+  SelectInput
+} from 'app/components/Form';
+import Option from './Option';
+import styles from '../surveys.css';
+import Icon from 'app/components/Icon';
+import { ConfirmModalWithParent } from 'app/components/Modal/ConfirmModal';
+import { mappings, QuestionTypes } from '../../utils';
+
+type Props = {
+  deleteQuestion: number => Promise<*>,
+  questionData: Object,
+  question: string,
+  index: number
+};
+
+const questionTypeToIcon = {
+  single_choice: 'radio-button-on',
+  multiple_choice: 'checkbox',
+  text_field: 'more'
+};
+
+const QuestionTypeOption = (props: Object) => (
+  <div
+    className={props.className}
+    onMouseDown={() => {
+      event.preventDefault();
+      event.stopPropagation();
+      props.onSelect(props.option, event);
+    }}
+    onMouseEnter={() => props.onFocus(props.option, event)}
+    onMouseMove={() => {
+      if (props.isFocused) return;
+      props.onFocus(props.option, event);
+    }}
+  >
+    <Icon
+      name={questionTypeToIcon[props.option && props.option.value]}
+      style={{ marginRight: '15px' }}
+    />
+    {props.children}
+  </div>
+);
+
+const QuestionTypeValue = (props: Object) => (
```

Comment:
What's the difference between `QuestionTypeValue` and `QuestionTypeOption`? What are the `optionComponent` and `valueComponent` passed to `SelectInput.Field`?

Reply:
The `Value`-one is for the selected value that is always visable. The `Option` is for the dropdown-list. More or less copied from the  "Custom Placeholder, Option, Value, and Arrow Components" section from https://jedwatson.github.io/react-select/

Reply:
Should be possible to reduce the amount of reused code though, surely. Is it significant that one has a `<span>` while the other doesn't? If it's just for the classname we can send both that and the surrounding div's classname as props and put the rest in a shared generator-component

Reply:
Yeah, that is possible! The hardcoded classNames are required tho. They are special for `react-select`